src/charm.py

```diff
@@ -30,105 +21,63 @@ class CephFSServerProxyCharm(ops.CharmBase):
 
     _REQUIRED_CONFIGS = ["fsid", "sharepoint", "monitor-hosts", "auth-info"]
 
-    def __init__(self, *args, **kwargs):
+    def __init__(self, *args, **kwargs) -> None:
         super().__init__(*args, **kwargs)
-        self._cephfs_share = CephFSProvides(self, "cephfs-share")
+        self._filesystem = FilesystemProvides(self, "filesystem", "server-peers")
         self.framework.observe(self.on.config_changed, self._on_config_changed)
-        self.framework.observe(self._cephfs_share.on.share_requested, self._on_share_requested)
 
     def _on_config_changed(self, _) -> None:
         """Handle updates to CephFS server proxy configuration."""
-        if self.get_state("config"):
-            logger.warning("Filesystem info can only be set once. Ignoring new config.")
-            return
-
         config = {k: self.config.get(k) for k in CephFSServerProxyCharm._REQUIRED_CONFIGS}
 
         # This method catches both uninitialized configs and empty strings/lists.
         missing = [k for (k, v) in config.items() if not v]
 
         if missing:
-            msg = f"Missing config key{'s' if len(missing) > 1 else ''} {', '.join(missing)}"
+            if len(missing) > 1:
+                fmt = "Missing config keys {}"
+            else:
+                fmt = "Missing config key {}"
+
+            msg = fmt.format(", ".join(missing))
```

Comment:
Hmm... not sure how I feel about have 6 LoC just for creating a status message :sweat_smile: 

Perhaps we could format the message in such a way where the missing config keys are presented as an array in the status message:

```suggestion
            msg = f"Missing required configuration for {', '.join(f'`{k}`' for k in missing)}")
```